implementation/src/main/java/io/smallrye/health/SmallRyeHealthReporter.java

```diff
@@ -69,9 +95,13 @@
     @ConfigProperty(name = "io.smallrye.health.emptyChecksOutcome", defaultValue = "UP")
     String emptyChecksOutcome;
 
-    private List<HealthCheck> additionalChecks = new ArrayList<>();
+    private final List<HealthCheck> additionalChecks = new ArrayList<>();
```

Comment:
You could have a single list if every time you add a synchronous additional checks you wrap is as an async check.